roles/disaster_recovery/tasks/clean/remove_valid_filtered_master_domains.yml

```diff
@@ -2,23 +2,37 @@
 - name: Remove valid storage domain main block
   block:
     - name: Fetch active/maintenance/detached storage domain for remove
+      ovirt_storage_domain_info:
+        pattern: >
```

Comment:
I must admit I don't have a lot of Ansible experience, but could some playing with `()` be a more simple option to fix the original code instead of splitting the code into 3 tasks with some amount of duplicate code?
```
        pattern: >
          name={{ storage['dr_' + dr_source_map + '_name'] }} and
          (
            (
              datacenter={{ storage['dr_' + dr_source_map + '_dc_name'] }} and
              (
                {{ dr_active_domain_search }} or {{ dr_maintenance_domain_search }}
              )
            )
            or
            {{ dr_unattached_domain_search }}
          )
```

Reply:
> I must admit I don't have a vast Ansible experience, but could some "playing" with `()` be a more simple option to fix the original code instead of splitting the code into 3 tasks with some amount of duplicate code? Not sure whether the following syntax will work:
> 
> ```
>         pattern: >
>           name={{ storage['dr_' + dr_source_map + '_name'] }} and
>           (
>             (
>               datacenter={{ storage['dr_' + dr_source_map + '_dc_name'] }} and
>               (
>                 {{ dr_active_domain_search }} or {{ dr_maintenance_domain_search }}
>               )
>             )
>             or
>             {{ dr_unattached_domain_search }}
>           )
> ```
> 
> If not then let's leave your solution with 3 separate tasks.

Thanks for taking a look at the changes. I would have preferred to have one common condition as well. However, for readability and ease of maintenance sake, I feel my new approach is much simpler. I hope that works for you as well.